src/store/indexeddb.js

```diff
@@ -88,24 +107,31 @@ IndexedDBStore.prototype.startup = function() {
     if (this.startedUp) {
         return q();
     }
+    // FIXME: This is a bit dumb: we pull the sync data over from
+    // the web worker to then immediately push it back again without
+    // using it.
     return this.backend.connect().then(() => {
         return q.all([
-            this.backend.loadUsers(),
+            this.backend.loadUserPresenceEvents(),
             this.backend.loadAccountData(),
             this.backend.loadSyncData(),
         ]);
     }).then((values) => {
-        const [users, accountData, syncData] = values;
+        const [userPresenceEvents, accountData, syncData] = values;
         console.log(
             "Loaded data from database: sync from ", syncData.nextBatch,
             " -- Reticulating splines...",
         );
-        users.forEach((u) => {
+        userPresenceEvents.forEach(([userId, rawEvent]) => {
+            const u = new User(userId);
+            if (rawEvent) {
+                u.setPresenceEvent(new MatrixEvent(rawEvent));
+            }
+            this._userModifiedMap[u.userId] = u.getLastModifiedTime();
             this.storeUser(u);
         });
         this._syncTs = Date.now(); // pretend we've written so we don't rewrite
-        this.setSyncToken(syncData.nextBatch);
```

Comment:
See https://github.com/matrix-org/matrix-js-sdk/pull/412/commits/0594a8d03a939df1e379013f53a71e8b095bbcae for why this is correct